docs/tutorials/python/tutorial_scripts/wiki.py

```diff
@@ -0,0 +1,284 @@
+    sub_wiki_1.id == retrieved_wiki.id,
+    sub_wiki_1.title == retrieved_wiki.title,
+]
+print(f"All fields match: {all(comparisons)}")
+
+# Section 2: WikiPage Markdown Operations
+# Create wiki page from markdown text
+markdown_content = """# Sample Markdown Content
+
+## Section 1
+This is a sample markdown file with multiple sections.
+
+## Section 2
+- List item 1
+- List item 2
+- List item 3
+
+## Code Example
+```python
+def hello_world():
+    print("Hello, World!")
+```
+"""
+
+# Create wiki page from markdown text
+sub_wiki_2 = WikiPage(
+    owner_id=my_test_project.id,
+    parent_id=root_wiki_page.id,
+    title="Sub Page 2 created from markdown text",
+    markdown=markdown_content,
+).store()
+
+# Create a wiki page from a markdown file
+# Create a temporary markdown gzipped file from the markdown_content
+markdown_file_path = "temp_markdown_file.md.gz"
+with gzip.open(markdown_file_path, "wt", encoding="utf-8") as gz:
+    gz.write("This is a markdown file")
+
+# Create wiki page from markdown file
+sub_wiki_3 = WikiPage(
+    owner_id=my_test_project.id,
+    parent_id=root_wiki_page.id,
+    title="Sub Page 3 created from markdown file",
+    markdown=markdown_file_path,
+).store()
+
+# Download the markdown file
+# delete the markdown file after uploading to test the download function
+os.remove(markdown_file_path)
+# Note: If the markdown is generated from plain text using the client, the downloaded file will be named wiki_markdown_<wiki_page_title>.md.gz. If it is generated from an existing markdown file, the downloaded file will retain the original filename with the .gz suffix appended.
+# Download the markdown file for sub_wiki_2 that is created from markdown text
+wiki_page_markdown_2 = WikiPage(
+    owner_id=my_test_project.id, id=sub_wiki_2.id
+).get_markdown(
+    download_file=True,
+    download_location=".",
+    download_file_name=f"wiki_markdown_{sub_wiki_2.title}.md.gz",
+)
+print(
+    f"Wiki page markdown for sub_wiki_2 successfully downloaded: {os.path.exists(f'wiki_markdown_{sub_wiki_2.title}.md.gz')}"
+)
+# clean up the downloaded markdown file
+os.remove(f"wiki_markdown_{sub_wiki_2.title}.md.gz")
+
+# Download the markdown file for sub_wiki_3 that is created from a markdown file
+wiki_page_markdown_3 = WikiPage(
+    owner_id=my_test_project.id, id=sub_wiki_3.id
+).get_markdown(
+    download_file=True, download_location=".", download_file_name=markdown_file_path
+)
+print(
+    f"Wiki page markdown for sub_wiki_3 successfully downloaded: {os.path.exists(markdown_file_path)}"
+)
+# clean up the downloaded markdown file
+os.remove(markdown_file_path)
+
+# Section 3: WikiPage with Attachments
+# Create a temporary file for the attachment
+attachment_file_name = "temp_attachment.txt.gz"
+with gzip.open(attachment_file_name, "wt", encoding="utf-8") as gz:
+    gz.write("This is a sample attachment.")
+
+# reformat '.' and '_' in the attachment file name to be a valid attachment path
```

Comment:
It's a bit unfortunate that we have to do this client side. I wonder if there is a better way that we can do this though. Maybe a static function on WikiPage that we can point to.

Is it possible that we take the string from the `markdown` field and automatically convert these? I imagine that `.` and `_` are not the only characters that have to be escaped.